src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -99,15 +99,23 @@ private IntPtr AllocException(Exception ex)
         [DllImport("jitinterface")]
         private extern static char* GetExceptionMessage(IntPtr obj);
 
+#if READY_TO_RUN
+        private ReadyToRunCodegenCompilation _compilation;
+#else
         private Compilation _compilation;
+#endif
         private JitConfigProvider _jitConfig;
 
         public CorInfoImpl(Compilation compilation, JitConfigProvider jitConfig)
         {
             //
             // Global initialization
             //
+#if READY_TO_RUN
+            _compilation = (ReadyToRunCodegenCompilation)compilation;
```

Comment:
Nit - I wonder whether it might be cleaner to make the ctor signature reflect the actual type - I believe the call sites can easily adapt to this.

Reply:
Fixed